tests/rdf-connect/ldes-client.test.ts

```diff
@@ -2989,4 +2981,927 @@ describe("Functional tests for the js:LdesClient RDF-Connect processor", () => {
         // Expect output stream to be closed
         expect(finished).toBeTruthy();
     });
+
+    test("Fetching an LDES unordered and checking if state is saved and enforced upon resume", async () => {
+        const outputStream1 = new SimpleStream<string>();
+
+        outputStream1.data((record) => {
+            // Check presence of SDS terms
+            expect(record.indexOf(SDS.stream)).toBeGreaterThan(0);
```

Comment:
If a bug would get introduced preventing the data to be outputted to the stream, this test will not catch this.
We can prevent this by using `expect.assertions(X);` with X the amount of expected `expect` calls, or with a `count` variable like in the already existing tests.
Comment holds for all the new tests.

Reply:
In these new tests there is a check for the total number of members emitted using the new `Client.memberCount` property, so essentially does the same as the local count in the older tests.

Reply:
Okay, good, didn't catch that one.
Let's maybe add a test then which tests that `Client.memberCount` matches the amount of actually emitted members.

Reply:
Added that extra check here https://github.com/rdf-connect/ldes-client/pull/94/commits/6e95279ecc204f7cf2d1dfa77ed9e2cb3500a75b